pkg/dmaasbackup/periodic_backup_test.go

```diff
@@ -0,0 +1,602 @@
+				Result(),
+			existingBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+				generateBackup("2020-06-20 06:00:00", "b1"),
+			},
+			expectedVeleroSchedule: []v1alpha1.VeleroScheduleDetails{
+				veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false),
+				veleroScheduleDetails(t, "2020-06-20 06:10:00", v1alpha1.Deleted, false),
+				veleroScheduleDetails(t, "2020-06-20 06:00:00", v1alpha1.Deleted, false),
+			},
+			expectedBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+				//	generateBackup("2020-06-20 06:00:00", "b1"),
+			},
+		},
+		{
+			name: "should update last schedule status as Erased if all backups are deleted",
+			dmaasbackup: dbkpbuilder(1).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:10:00", v1alpha1.Deleted, false)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:00:00", v1alpha1.Deleted, false)).
+				Result(),
+			existingBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+			},
+			expectedVeleroSchedule: []v1alpha1.VeleroScheduleDetails{
+				veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false),
+				veleroScheduleDetails(t, "2020-06-20 06:10:00", v1alpha1.Deleted, false),
+				veleroScheduleDetails(t, "2020-06-20 06:00:00", v1alpha1.Erased, false),
+			},
+			expectedBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+			},
+		},
+		{
+			name: "should not delete backups for schedule if schedule is not deleted",
+			dmaasbackup: dbkpbuilder(1).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:10:00", v1alpha1.Deleted, false)).
+				WithVeleroSchedules(veleroScheduleDetails(t, "2020-06-20 06:00:00", "", false)).
+				Result(),
+			existingBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+				generateBackup("2020-06-20 06:00:00", "b1"),
+			},
+			expectedVeleroSchedule: []v1alpha1.VeleroScheduleDetails{
+				veleroScheduleDetails(t, "2020-06-20 06:20:00", v1alpha1.Active, false),
+				veleroScheduleDetails(t, "2020-06-20 06:10:00", v1alpha1.Deleted, false),
+				veleroScheduleDetails(t, "2020-06-20 06:00:00", "", false),
+			},
+			expectedBackups: []velerov1api.Backup{
+				generateBackup("2020-06-20 06:20:00", "b1"),
+				generateBackup("2020-06-20 06:10:00", "b1"),
+				generateBackup("2020-06-20 06:00:00", "b1"),
+			},
+		},
+		{
+			name: "should not delete backup if creation of deletionrequest failed",
```

Comment:
is this testcase related to velero?
if I'm not wrong, dmaas-operator send deleteRequest, and velero supposed to delete the backup schedule if if fails.
But for unit testing, this PR should have faked velero behavior.

Reply:
For cleanup part we have created velero behavior for deletion of backup. This test case is to verify failure in backupdeleterequest during cleanup. This is related to velero, but required here since backup deletion happens through backuprequest and we are taking backup as input.